test/L2GovernorMetadata.t.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Test} from "forge-std/Test.sol";
+
+import {L2GovernorMetadata} from "src/L2GovernorMetadata.sol";
+import {Constants} from "test/Constants.sol";
+
+contract L2GovernorMetadataTest is Test, Constants {
+  L2GovernorMetadata l2GovernorMetadata;
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl("polygon_mumbai"));
+    l2GovernorMetadata = new L2GovernorMetadata(wormholeCoreMumbai);
+    l2GovernorMetadata.registerApplicationContracts(
+      6, bytes32(uint256(uint160(address(0x628C44d859d17aD932960DFcE226F5de427f9d6D))))
+    );
+  }
+}
+
+contract ReceiveEndcodedMsg is L2GovernorMetadataTest {
+  function testFork_CorrectlyStoreMetadata() public {
+    // Encoded message was generated from transaction
+    // 0x873dab653e1a8a746a8e4f51eb0e1de46f878255312e18506301e3cc5cb29954
+    // on Avalanche Fuji
+    l2GovernorMetadata.receiveEncodedMsg(
```

Comment:
Should I assume we are just setting the harness up here, and that more robust unit testing can/will be added later?

Reply:
Yes, in depth testing will be in a separate pr